lib/src/command/lish.dart

```diff
@@ -138,6 +141,32 @@ class LishCommand extends PubCommand {
     }
   }
 
+  Future<void> _publish(List<int> packageBytes) async {
+    try {
+      if (tokenStore.hasCredential(server)) {
+        // If there's a saved credential for the server, publish using
+        // httpClient which should authenticate with the server using
+        // AuthenticationClient.
+        await withAuthenticatedClient(cache, server, (client) {
+          return _publishUsingClient(packageBytes, client);
+        });
+      } else {
+        // If user had not yet logged in into the server, use OAuth2 credentials
+        // for authentication.
+        await oauth2.withClient(cache, (client) {
```

Comment:
> @jonasfj should we limit this to pub.dartlang.org/pub.dev?

I think some third party servers uses google id token to authenticate publish commands. So, limiting domain might cause incompatibility with those servers. But In other case this will add additional security / privacy (since id_token includes identifiers: email, id) layer 🤔

Reply:
Third-party servers should never be given access to the users `id_token` as created by `oauth2.dart`.

That `id_token` is intended for use with `pub.dev`, hence, a third-party server would possibly be able to use this to make requests to pub.dev on behalf of a users. This would be a security vulnerability.

Reply:
But AFAIK it's currently possible for 3rd party servers to know id_tokens cause when I did my research prior to working on this PR I found out that exists pub clones are authenticating using pub.dev tokens when uploading archives. You can check out [here](https://github.com/bytedance/unpub/blob/master/unpub/lib/src/app.dart#L79-L101) for example.